language/vm/vm_runtime/src/runtime.rs

```diff
@@ -109,11 +109,19 @@ impl<'alloc> VMRuntime<'alloc> {
     /// transaction output.
     pub fn execute_block_transactions(
         &self,
-        txn_block: Vec<SignedTransaction>,
+        txn_block: Vec<Transaction>,
         data_view: &dyn StateView,
     ) -> Vec<TransactionOutput> {
+        let mut txns = vec![];
+        for txn in txn_block {
+            match txn {
+                Transaction::UserTransaction(user_txn) => txns.push(user_txn),
```

Comment:
I think it's better to do `txn.as_signed_user_txn().unwrap().clone()` to guarantee only user_txns are passed in for now. 

(or add `impl TryFrom<Transaction> for SignedTransaction` temporarily if you really don't like the `.clone()`)